### ISSUE TITLE
Search : URL user or URL status doesn't work 

### ISSUE THREAD

**Issue by koreus** [mastodon/mastodon#23927]
### Steps to reproduce the problem

Hi,

When I tried to search « URL user » or  « URL status » (URL from a small instance) it returns 0 result.

Example with URL status :

1. Go on [mastodon.social](https://mastodon.social/)
2. Search this status : https://koreus.social/@koreus/109954359223789780
3. It returns 0 result : Could not find anything for these search terms

Another example with a URL user

1. Go on [mastodon.social](https://mastodon.social/)
2. Search this user  : https://koreus.social/@koreus
3. It returns 0 result : Could not find anything for these search terms
4. Try to search : @koreus@koreus.social : It works !

Thx for you help

### Expected behaviour

It should load a user or status

### Actual behaviour

It returns no result

### Specifications

Mastodon 4.1.0

**Comment by TheDevJoao** [comment]
Hey @koreus!

I replicated the issue and it doesn't appear to be a bug. Both your account and status belong to your server instance "koreus", right? When searching using a URL from mastodon.social's search bar, the results are limited to mastodon.social's server domain. 

I created an account that belongs to mastodon.social and am able to search for both my account profile and status:

Searching for my user: https://mastodon.social/@testdevguy

![image](https://user-images.githubusercontent.com/64037198/231525989-79cadcc8-1699-4138-9bb8-66d6749d3193.png)

Searching for this status: https://mastodon.social/@testdevguy/110186832802645269

![image](https://user-images.githubusercontent.com/64037198/231526391-c8536e70-066b-4d1e-8626-a131202433d5.png)

If you click the search bar from your home directory you should get the following prompt:

![image](https://user-images.githubusercontent.com/64037198/231525229-bbd15bc4-3f1b-43ce-b38e-8394a029ed99.png)

Basically if you want to find users from other servers you have to specify the username + server domain like you did in your 4th step if they belong to an instance that allows external communication. 😄

**Comment by kzxpr** [comment]
Has it always been this way?
I think I've used URLs to search for posts that were from another instance before. Like if you press "Copy link to post" on a external post and paste it in the search bar, it showed up.
I'm not 100% sure, but I think this would be an essential feature - especially if servers can't fetch old posts from external accounts.

**Comment by kzxpr** [comment]
@Gargron Are URLs supposed to trigger a general ActivityPub-lookup? Because then I can't get it to work on either of the Mastodon instances I'm registered on (both use v4.1.2). The same URL works on my CalcKey.

Here's an example of a URL I've tested with:
https://kasperaliteten.dk/u/k/statuses/07b7e4713b045a49a4b55bc8b4d5628f
